### DISCUSSION TITLE
New version of my book "A Lisp Programmer Living in Python-Land: The Hy Programming Language"

### DISCUSSION THREAD

**Discussion by mark-watson** [hylang/hy#2443] [Show and tell]
I just released a new version of my book A Lisp Programmer Living in Python-Land: The Hy Programming Language (updated code to Hy version 0.26.0, added chapters for OpenAI APIs and LangChain). Purchase or read free online: https://leanpub.com/hy-lisp-python
There are three major changes:
1. Updated all code examples to use latest Hy language version 0.26.0
2. Added a chapter with examples using Open AI GPT-3.5 APIs
3. Added a chapter using the LangChain library
The Hy language is very cool. It is a Lisp language with a syntax similar to Clojure. Hy runs on top of Python so all Python libraries and frameworks are available.

**Comment by Kodiologist** [comment]
I belatedly noticed the existence of your book some time ago, and I was a bit surprised that you haven't really been in contact with us, since we're a tiny community. But if you didn't find any new bugs in the process of writing an entire book on the language, then that's good news as far as I'm concerned.

I know the journey to Hy 1.0 has been sluggish at best, but at least I don't expect to make a great deal more breaking changes.

Do you notice any particular deficiencies in the official documentation while writing your book?

**Reply by mark-watson** [reply]
No real problems with the language. It took a short while to absorb the language changes for 0.26.0, but it was all good after that.

I seldom refer to the documentation now since I have written a fair amount of Hy code.

I don't cover editors in the book, but I use hy-mode and Emacs and it took a while to get things working smoothly. I ended up disabling all auto-completions except for the GitHub Copilot Emacs support, so as I am editing I depend on Copilot for auto-completion. BTW, Copilot does a fairly good job with Hy, when you are editing an existing source file.

**Reply by mark-watson** [reply]
There is one thing I didn’t see in the documentation, but maybe I just missed it: examples with the new syntax for optional function arguments.

**Reply by Kodiologist** [reply]
The latest version of the tutorial and the help entry for `defn` have that.
